### DISCUSSION TITLE
Move an app window from one space to another with keyboard shortcut

### DISCUSSION THREAD

**Discussion by MissingJoe** [rxhanson/Rectangle#1354] [Ideas]
I usually work with many spaces, so it's very nice for me to add features of moving apps from one space to another with keyboard shortcut. If so, I won't open the mission control and drag the windows to another space, it's annoying.  I read some docs on macOS, but it seems that this keyboard shortcut features don't exist in macOS. Is there any other way to solve this problem?

**Comment by rxhanson** [comment]
Apple doesn't provide a simple API for doing this, so any app that lets you do it is using sort of an input hack (or private framework usage) to accomplish it. I prefer to not add hacky things into Rectangle because it is supposed to be simple functionality that just works all the time.

I did add it into [Rectangle Pro](https://rectangleapp.com/pro) (paid), which lets you configure keyboard shortcuts for moving a window next/prev space. It uses the input hack (and does not use private frameworks at all). It works for nearly all the apps I use, but sometimes can still be a little finicky with very specific apps and unusual or fast space switching timing.

**Comment by lasselaakkonen** [comment]
Hi @rxhanson, I couldn't find the keyboard shortcuts for moving a window next/prev space in the Pro app, does it still exist or did Sequoia take it away?

**Reply by rxhanson** [reply]
![image](https://github.com/user-attachments/assets/709526f9-3908-4e1d-8f7a-b8867409cf89)

**Reply by lasselaakkonen** [reply]
@rxhanson I was blind. Thanks! You got a new paying customer!
